storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -4010,6 +3994,32 @@ dberr_t fil_space_t::garbage_collect(bool shutdown)
     return err;
   }
 
+  /* Remove cached undo log segments in the rollback segment
+  header page exist in system tablespace */
+  mtr.start();
+  mtr.x_lock_space(fil_system.sys_space);
+  for (ulint rseg_id= 0; rseg_id < TRX_SYS_N_RSEGS; rseg_id++)
+  {
+    trx_rseg_t *rseg= &trx_sys.rseg_array[rseg_id];
+    if (rseg->space->id == 0)
+    {
+      dberr_t err= DB_SUCCESS;
+      buf_block_t *block=
+        buf_page_get_gen(page_id_t{0, rseg->page_no}, 0,
+                         RW_X_LATCH, nullptr, BUF_GET, &mtr, &err);
```

Comment:
It is not necessary to initialize `err` in the caller. It will be initialized in every code path of `buf_page_get_gen()`.